### ISSUE TITLE
Develop a criteria for work and difficulty estimation

### ISSUE THREAD

**Issue by KOLANICH** [open-source-ideas/open-source-ideas#55]
For now the issues in this repo are marked with ```Little work```, ```Medium work```, ```Much work``` and ```Beginner```, ```Intermediate```, ```Advanced``` labels.

But what is `Much work` ? Is developing an app in the way "take existing libraries, make them do their tasks and return the results, compare the results and show them to user" really so much work for an advanced engineer? I guess we need a more objective criteria of amount and complexity of the work, taking in account
* existing libraries
* libraries to be created
* development and reverse-engineering tools and their capabilities
* amount of research and requirements to the researchers (I mean here not level, but domain and specialization, if the work requires expert knowlege in different domains, it is more complex and may require more people (with competencies in differrent domains) to be finished).

**Comment by FredrikAugust** [comment]
Hey, I guess this is directed mostly at me, since I've been going over all the issues and filling in these labels as I saw fit. 

The problem here is that there isn't any real way to measure the difficulty of a programming task, and it varies radically depending on the person and his/her background. Of course; a simple web app will be easy to develop for an experienced web dev, but it might be very hard for a new web dev or a non-web dev.

I presume the issue you're referring to here is [the RPC wrapper library](https://github.com/open-source-ideas/open-source-ideas/issues/49) which I marked as `Advanced` and `Much work`. This might look easy to you, if you have a lot of experience with meta-programming, other RPC libs etc., but to me at least, _this_;

> The library should add an import hook and have a set of modules imported through that lib. When a the hook is activated, it checks if the name asked to be imported is in the set, and if it is, it starts the backend and communicates to it via a shared memory page (in the case of local one) or via network. All the communication must be authenticated and optionally encrypted. The backend imports the library and allows the frontend which have started it to remotely execute code by sending special commands in a custom binary protocol. This protocol must have some comands to get values of variables.

doesn't look very easy.

A possible solution is to completely remove the _difficulty_ rating we have now, and replace it with tags specifying the tech field; technologies, programming languages etc. So your RPC idea might have the tags `RPC`, `python` bla. bla.

This would also remove the biases regarding difficulty from the people setting it (aka me).

Furthermore, perhaps we should consider extending the submission _form_ to include some of the information you provided in this issue, ie. libraries, knowledge required (more specific than "very experienced in x language"), and estimated dev time (`Much work` isn't very descriptive).

**Comment by KOLANICH** [comment]
>I presume the issue you're referring to here is the RPC wrapper library which I marked as Advanced and Much work

In fact it was about #54 : we have a glue language python, we have libraries measuring speed from speedtest in python, we have bindings of libtorrent rasterbar to python.  Though I have not tried that (somI don't know if any obstacles like missing parts of api), but it may be easy to create a PoC: to do it all that is needed to be done is to bring everything together. The prototype should be immediately usuful. All other points will be complications, but when one has a working prototype, he can start gaining community.

**Comment by FredrikAugust** [comment]
I see. I'd still argue it's a lot of work, which brings me back to my comment about reworking the label system. As this might look simple to you, but tedious to another dev.

**Comment by KOLANICH** [comment]
>A possible solution is to completely remove the difficulty rating we have now, and replace it with tags specifying the tech field; technologies, programming languages etc.

Yes, but I guess only fields requiring domain knowlege you cannot gain in an hour of reading should be used for that. I mean the requirements creating entry barriers. For example it's pretty easy to start working with xgboost if you are familiar with basics of machine learning, but working with neural networks require much more domain knowledge and hardware resources. Maybe better labels should be not describing skill level, but barriers like "modern GPU", "neural networks", "lot of time", "category theory", "numerical simulation"

**Comment by FredrikAugust** [comment]
Look at #71. As of now it's really the only solution I think is feasible to implement, any other would (I think) simply be demoralizing for the one creating the issue, as it requires them to write a lot.

**Comment by mikaelbr** [comment]
Moved to https://github.com/open-source-ideas/meta/issues/6
